api/src/main/java/org/openmrs/util/ConceptReferenceRangeUtility.java

```diff
@@ -0,0 +1,121 @@
+package org.openmrs.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.joda.time.LocalTime;
+import org.openmrs.Concept;
+import org.openmrs.Obs;
+import org.openmrs.Person;
+import org.openmrs.api.context.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * A utility class that evaluates the concept ranges 
+ * 
+ * @since 2.7.0
+ */
+public class ConceptReferenceRangeUtility {
+	
+	private static final Logger logger = LoggerFactory.getLogger(ConceptReferenceRangeUtility.class);
+	
+	public ConceptReferenceRangeUtility() {
+	}
+	
+	/**
+	 * This method evaluates the given criteria against the provided person/patient.
+	 *
+	 * @param criteria the criteria string to evaluate e.g. "$fn.getAge($patient, 'YEARS') > 1 && $fn.getAge($patient, 'YEARS') < 10"
+	 * @param person person object containing attributes to be used in the criteria
+	 *                  
+	 * @return true if the criteria evaluates to true, false otherwise
+	 * 
+	 * @since 2.7.0
+	 */
+	public boolean evaluateCriteria(String criteria, Person person) {
+		if (person == null) {
+			throw new IllegalArgumentException("Failed to validate with reason: patient is null");
+		}
+		
+		if (StringUtils.isEmpty(criteria)) {
+			throw new IllegalArgumentException("Failed to validate with reason: criteria required");
+		}
+		
+		VelocityContext velocityContext = new VelocityContext();
+		velocityContext.put("fn", new ConceptReferenceRangeUtility());
+		
+		velocityContext.put("patient", person);
+		
+		VelocityEngine velocityEngine = new VelocityEngine();
+		
+		StringWriter writer = new StringWriter();
+		String wrappedCriteria = "#set( $criteria = " + criteria + " )$criteria";
+		velocityEngine.evaluate(velocityContext, writer, ConceptReferenceRangeUtility.class.getName(), wrappedCriteria);
+		
+		String evaluatedCriteria = writer.toString();
+		
+		return Boolean.parseBoolean(evaluatedCriteria);
+	}
+	
+	/**
+	 * Gets the latest Obs by concept.
+	 *
+	 * @param conceptRef can be either concept uuid or conceptMap's code and sourceName 
+	 *                   e.g "bac25fd5-c143-4e43-bffe-4eb1e7efb6ce" or "CIEL:1434"
+	 *                   
+	 * @return Obs
+	 * 
+	 * @since 2.7.0
+	 */
+	public Obs getLatestObsByConcept(String conceptRef) {
+		Concept concept = Context.getConceptService().getConceptByReference(conceptRef);
+
+		if (concept != null) {
+			List<Obs> observations = Context.getObsService().getObservations(
+				null, 
+				null, 
+				Collections.singletonList(concept), 
+				null, 
+				null, 
+				null, 
+				null, 
+				null, 
+				null,
+				null, 
+				null, 
+				false
+			);
+
+			// Return the latest Obs by sorting the list by dateCreated
+			return observations.stream()
+				.max(Comparator.comparing(Obs::getDateCreated))
```

Comment:
The getObservations() method has a `sort` parameter. Can you use that instead?

Reply:
Alternatively you could just use the new method that you added to the ObsService.

Reply:
Let me use `sort` since I have already removed the unused method in ObsService as advised earlier .

Reply:
How will this be faster than asking the database to do the sorting?

Reply:
I meant instead of using stream's `sorted()` method. I see what @dkayiwa  meant. Let me update.